Remove temporary files created in EDS tests

erlend-aasland opened the pull request:
Clean-up post #474

sveinse reviewed (approved):
Looks good

erlend-aasland commented on test/test_eds.py:
```diff
@@ -222,6 +222,7 @@ def test_export_eds_unknown_doctype(self):
         import io
         filelike_object = io.StringIO()
         self.addCleanup(filelike_object.close)
+        self.addCleanup(os.unlink, "filename")
```
Note: this will fail if the first iteration of the subsequent test loop fails; no file named `filename` is created, thus `FileNotFoundError` is raised. This exposes another bug of `export_od`, namely that it creates the file on disk before actually checking if a valid format was provided.

sveinse replied:
🎉  Isn't that why test cases exists? To find bugs I mean?

erlend-aasland replied:
I created #485 for tracking this.

sveinse commented:
I wanted to merge this, but I suddenly got stage fright: @acolomb what is the practice in canopen wrt. merging vs squash and merge? (I'm sorry that this Q is OT to the PR, do we have any forum for discussing best practices and policies for dev in canopen?)

erlend-aasland commented:
> I wanted to merge this, but I suddenly got stage fright: @acolomb what is the practice in canopen wrt. merging vs squash and merge?

The Git log shows that merge commits have not been committed the last three years; squashed commits seems to be the prevailing style now, which I 100% fully support. Squashed commits with well-written commit titles/bodies makes for a readable Git history and increased maintainability.

acolomb commented:
> (I'm sorry that this Q is OT to the PR, do we have any forum for discussing best practices and policies for dev in canopen?)

Probably best to use GitHub's discussion tool for this.  But with the currently small base of active contributors, it's not tragic to ask some OT questions within a PR :-)

erlend-aasland commented:
Following up in the OT track 😄 Is there an instant-messaging forum for CAN/CANopen stuff (Discord or similar)?

sveinse commented:
Squash merging is an opinionated 🔥  topic . Some love it, other loathe it, so I had to ask. What is "obviously, my friend" one place might not be that in other projects. YMMV.

sveinse commented:
> Following up in the OT track 😄 Is there an instant-messaging forum for CAN/CANopen stuff (Discord or similar)?

I don't know of any, but I'm in 😎

acolomb commented:
In general, this project is not as professionally managed, with all kinds of supporting infrastructure, as maybe other, bigger projects.  I'm very grateful for all your contributions lately, @erlend-aasland and @sveinse.  But I think time is better spent looking at the code and discussing asynchronously via GitHub discussions / issues / PRs, than on setting up some kind of developer hangout.  I for one wouldn't have the opportunity to participate there regularly, thus working through notifications as time permits is my preferred workflow.

erlend-aasland commented:
Anyway, let me know if you want any other changes before landing this :)

sveinse commented:
My first merge as maintainer in canopen, so I'm open to feedback if I did anything wrong @acolomb

erlend-aasland commented:
Congrats on your first merge! I'm curious why you chose a merge commit:

- the PR consisted of two commits, the latter being totally uninteresting in a Git log, only serving to cluttering the Git history
- there has been no other merge commits the last three years; the de facto practice is squashed commits, as also confirmed by André
- three (!) commits (the two from this PR and the merge commit) seems overly verbose for a trivial change

sveinse commented:
Oh, damn! Then I actually did fail. My intentions were to do a squash merger and I thought I did, but evidently I did not do that. So that's the rookie mistake then. For that I'm very sorry. 😢  (For the record: I'm not a git or github rookie, just new in this project).

acolomb commented:
Life's not perfect. Neither is any project's Git history. Let's view it as a minor detail of such small importance that we can forget about it right away :-D

Sorry for not participating closely. Things got in the way this afternoon. But good to see we have one more co-maintainer to take care of stuff.
